controllers/internal/crutils/static_resource_generator.go

```diff
@@ -73,34 +73,20 @@ func (c *IBMBlockCSI) GenerateCSIDriver() *storagev1.CSIDriver {
 }
 
 func (c *IBMBlockCSI) GenerateControllerServiceAccount() *corev1.ServiceAccount {
-	secrets := []corev1.LocalObjectReference{}
-	if len(c.Spec.ImagePullSecrets) > 0 {
-		for _, s := range c.Spec.ImagePullSecrets {
-			secrets = append(secrets, corev1.LocalObjectReference{Name: s})
-		}
-	}
-
-	return &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      config.GetNameForResource(config.CSIControllerServiceAccount, c.Name),
-			Namespace: c.Namespace,
-			Labels:    c.GetLabels(),
-		},
-		ImagePullSecrets: secrets,
-	}
+	secrets := getImagePullSecrets(c.Spec.ImagePullSecrets)
+	return getServiceAccount(c, secrets, config.CSIControllerServiceAccount)
 }
 
 func (c *IBMBlockCSI) GenerateNodeServiceAccount() *corev1.ServiceAccount {
-	secrets := []corev1.LocalObjectReference{}
-	if len(c.Spec.ImagePullSecrets) > 0 {
-		for _, s := range c.Spec.ImagePullSecrets {
-			secrets = append(secrets, corev1.LocalObjectReference{Name: s})
-		}
-	}
+	secrets := getImagePullSecrets(c.Spec.ImagePullSecrets)
+	return getServiceAccount(c, secrets, config.CSINodeServiceAccount)
+}
 
+func getServiceAccount(c *IBMBlockCSI, secrets []corev1.LocalObjectReference,
+	serviceAccountResourceName config.ResourceName) *corev1.ServiceAccount {
```

Comment:
if we are sending `c` to `getServiceAccount`, why do we need `getImagePullSecrets` to be called out of `getServiceAccount`?
can we have:
```suggestion
func getServiceAccount(c *IBMBlockCSI,
	serviceAccountResourceName config.ResourceName) *corev1.ServiceAccount {
	secrets := getImagePullSecrets(c.Spec.ImagePullSecrets)
```

Reply:
done